packages/jest-cli/src/runJest.js

```diff
@@ -151,7 +152,7 @@ const runJest = async (
   pipe: stream$Writable | tty$WriteStream,
   testWatcher: TestWatcher,
   startRun: () => *,
-  onComplete: (testResults: any) => any,
+  onComplete: (testResults: AggregatedResult) => any,
```

Comment:
Note: since I updated this method to use the `AggregatedResult` type I had to loosen the requirements on that types properties due to [this](https://github.com/sandersky/jest/blob/eaa01e939d2865037d40ad6374691c2da0664ef3/packages/jest-cli/src/runJest.js#L179). This caused code in a few other places to require additional checks to ensure things aren't undefined before accessing/modifying them.